### PR TITLE
Return all the messages loaded in the Translator

### PR DESCRIPTION
**Pull request by pensiero**
Added public method to the I18n\Translator that return all the messages loaded in the Translator: useful in case we want to manage all of them with some tools (refactoring, find&replace, po2json).

### CODE REVIEW

File: test/Translator/TranslatorTest.php [hunk 1]
```diff
@@ -386,4 +386,20 @@ public function testListenerOnNoMessagesLoadedEventCanReturnTextDomainObject()
         $this->assertNull($doNotTrigger);
         $this->assertEquals('BOOYAH', $result);
     }
+
+    public function testGetAllMessagesLoadedInTranslator()
```
Comment (juriansluiman) [inline test/Translator/TranslatorTest.php:390]: I would say the only relevant test here is a) check if the result is an array when you supply it with the translation file and b) check if the size of the array equals the count of `$allMessages`.

Also, what happens when the text domain is not found or the locale does not exist? What is the return type and can you test those two cases as well?

**Comment by pensiero** [comment]
Thanks @juriansluiman, do you think it's good enough now?
I have created 3 test methods, do I have to put all the tests in a single method?

**Comment by juriansluiman** [comment]
@Gamempire this is what I meant yes. Putting them in three separate cases makes it distinct what you test and what could go wrong. The only thing we _could_ argue about it the name of the method. Now it is `testGetAllMessagesLoadedInTranslatorWithNotFoundTextDomain()`, or in plain English: "Get all messages loaded in translator with not found text domain". This is what the _implementation_ of the test looks like but it's not the thing you actually test.

Personally, I like more a descriptive title which can be used with a docx format having a cross in front or not - depending on the success or failure of the test. Therefore, test methods like `testTranslatorReturnsNullWhenGivenTextDomainIsNotFound()` is -imho- more appropriate (similar for the locale). 

Furthermore, :+1: for the work :)

**Comment by pensiero** [comment]
Very good suggestion, changed! :)

**Comment by Maks3w** [comment]
We had to rewrite master branch for match release-2.4.2. Could you rebase your work again?

Sorry for the inconvenience

**Comment by pensiero** [comment]
Is it ok now?

**Comment by Maks3w** [comment]
Please rebase for to clean the history. PR has more than 250 commits.

You can create a branch from the upstream master, cherry pick your commits and force push to your fork master

**Comment by pensiero** [comment]
I have created a new pull request from a new fork: https://github.com/zendframework/zend-i18n/pull/3
